### ISSUE TITLE
Error: 13 INTERNAL: <message> reported for exceptions thrown by user-provided stream call handler

### ISSUE THREAD

**Issue by kskalski** [grpc/grpc-node#1661]
### Problem description
My grpc-js client is receiving a streaming response, but seems to fail parsing it for this particular message/case, even though it works well on a wide variety of other messages.
The error is 
```
Error: 13 INTERNAL: Response message parsing error: Cannot read property 'Topics' of undefined
    at Object.callErrorFromStatus (D:\rep\Kogut\src\ElectronUI\node_modules\@grpc\grpc-js\build\src\call.js:31:26)
    at Object.onReceiveStatus (D:\rep\Kogut\src\ElectronUI\node_modules\@grpc\grpc-js\build\src\client.js:327:49)
    at Object.onReceiveStatus (D:\rep\Kogut\src\ElectronUI\node_modules\@grpc\grpc-js\build\src\client-interceptors.js:294:181)
    at D:\rep\Kogut\src\ElectronUI\node_modules\@grpc\grpc-js\build\src\call-stream.js:130:78
    at processTicksAndRejections (internal/process/task_queues.js:75:11) {
  code: 13,
  details: "Response message parsing error: Cannot read property 'Topics' of undefined",
  metadata: Metadata { internalRepr: Map(0) {}, options: {} }
}
```

### Reproduction steps
This is happening consistently for me on this message being sent to the client from server. I narrowed down that:
* the message is correctly delivered from server to client - I logged this on server (app code): 
```
 Writing collated size 1375 base64 ...
```
on the client (added some logging to call-stream.js):
```
2021-01-11T18:19:38.192Z | call_stream | [2] receive HTTP/2 data frame of length 1380
2021-01-11T18:19:38.193Z | call_stream | [2] parsed message of length 1380
2021-01-11T18:19:38.193Z | call_stream | [2] filterReceivedMessage of length 1380
2021-01-11T18:19:38.193Z | call_stream | [2] pushing to reader message of length 1375
2021-01-11T18:19:38.193Z | call_stream | [2] msg base64 ... (MATCHING BASE64 SAME AS IN SERVER)
2021-01-11T18:19:38.194Z | call_stream | [2] cancelWithStatus code: 13 details: "Response message parsing error: Cannot read property 'Topics' of undefined"
2021-01-11T18:19:38.194Z | call_stream | [2] ended with status: code=13 details="Response message parsing error: Cannot read property 'Topics' of undefined"
```

* this is not a parser bug, I used the same ``grpc.GrpcObject``, to get out the ``responseDeserialize`` method and feed it with the same base64 data buffer and it parses it correctly
* thus it seems to be happening only in context of stream-call, which got me stuck in debugging at this point

### Environment
 - OS name, version and architecture: Windows 10, x64
 - Node version: Electron 12.beta6
 - Node installation method: Electron
 - Package name and version:     "@grpc/grpc-js": "1.2.3",  "@grpc/proto-loader": "0.5.5",

**Comment by murgatroid99** [comment]
Have you reproduced this error in Node too, or just in Electron? Can you share some code that triggers this error?

**Comment by kskalski** [comment]
Only in Electron. I will try to reproduce it in a self-contained app, hopefully it's purely payload/method type related not dependent on a whole context of execution.

**Comment by kskalski** [comment]
Huh,  that error actually comes from my code, I just didn't notice that my handler could produce this kind of error message and was confused by the way grpc-js reported this.

I suppose things are fine, though maybe grpc-js could actually report a different error code / some additional message to notify that user provided handler thrown exception.

**Comment by murgatroid99** [comment]
You're saying that your server-side handler function is throwing an error with the message "Response message parsing error: Cannot read property 'Topics' of undefined"? I can look into adding to the message to indicate that it came from the handler function, but you should also avoid throwing errors in handlers in the first place. Instead you should use the callback for unary response methods, or emit an `error` event on the call for streaming response methods.

**Comment by kskalski** [comment]
Yap, just small clarifications:
- it's a client code (in main node-process of Electron)
- indeed my handler was throwing that error
- I think some prefix/hint on the message that it came from handler would be useful for anybody falling into this case
- indeed I didn't have any catch in my handler, so unfortunately I cut myself 
- I'm not sure about details of you suggestion, I did provide the 'error' handler by doing ``call.on('error', console.log);``, I suppose I could add some clarifying message here instead, which would help me diagnose the source of the error (I'm doing a bidirectional client call)

**Comment by murgatroid99** [comment]
OK, if this is happening in the client, that is entirely different from what I said, and in that case I don't know what is happening here. I am not aware of any code path that could turn an error thrown by client code into the kind of error you reported. Are you *sure* it is not an error from the message parsing code? Because that is what the original error seems to indicate.

**Comment by murgatroid99** [comment]
I see what happened here. There's an overly broad `try` block that is misinterpreting the error your code is throwing as parsing error, which is why it is adding the prefix "Response message parsing error" to your error. I can fix that, then your error will be thrown as normal.

**Comment by murgatroid99** [comment]
I've published a patch in version 1.2.5 that should fix this.

**Comment by kskalski** [comment]
Thanks
